app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaDisplayable.java

```diff
@@ -161,21 +161,26 @@ public boolean hasStore() {
   }
 
   public GridStoreMetaWidget.HomeMeta.Badge getBadge() {
-    switch (getPojo().getData()
-        .getStore()
-        .getBadge()
-        .getName()) {
-      case BRONZE:
-        return GridStoreMetaWidget.HomeMeta.Badge.BRONZE;
-      case SILVER:
-        return GridStoreMetaWidget.HomeMeta.Badge.SILVER;
-      case GOLD:
-        return GridStoreMetaWidget.HomeMeta.Badge.GOLD;
-      case PLATINUM:
-        return GridStoreMetaWidget.HomeMeta.Badge.PLATINUM;
-      case NONE:
-      default:
-        return GridStoreMetaWidget.HomeMeta.Badge.NONE;
+    if (hasStore()) {
+
+      switch (getPojo().getData()
+          .getStore()
+          .getBadge()
+          .getName()) {
+        case BRONZE:
+          return GridStoreMetaWidget.HomeMeta.Badge.BRONZE;
+        case SILVER:
+          return GridStoreMetaWidget.HomeMeta.Badge.SILVER;
+        case GOLD:
+          return GridStoreMetaWidget.HomeMeta.Badge.GOLD;
+        case PLATINUM:
+          return GridStoreMetaWidget.HomeMeta.Badge.PLATINUM;
+        case NONE:
+        default:
+          return GridStoreMetaWidget.HomeMeta.Badge.NONE;
+      }
+    } else {
+      return null;
```

Comment:
Would it become cleaner if the else return GridStoreMetaWidget.HomeMeta.Badge.NONE and NONE would return a new badge enum value TIN (or DEFAULT)?